projects/plugins/boost/app/admin/class-admin.php

```diff
@@ -84,12 +86,32 @@ public function enqueue_styles() {
 	public function enqueue_scripts() {
 		$internal_path = apply_filters( 'jetpack_boost_asset_internal_path', 'app/assets/dist/' );
 
+		$critical_css_gen_handle = 'jetpack-boost-critical-css-gen';
+
+		wp_register_script(
+			$critical_css_gen_handle,
+			plugins_url( $internal_path . 'critical-css-gen.js', JETPACK_BOOST_PATH ),
+			array(),
+			JETPACK_BOOST_VERSION,
+			true
+		);
+
 		$admin_js_handle = 'jetpack-boost-admin';
 
+		$admin_js_dependencies = array(
+			'wp-i18n',
+			'wp-components',
+		);
+
+		// Enqueue the critical CSS generator script if Critical CSS is enabled.
+		if ( ( new Modules_Index() )->is_module_enabled( Critical_CSS::get_slug() ) ) {
+			$admin_js_dependencies[] = $critical_css_gen_handle;
+		}
```

Comment:
As discussed in team call, it should check if Critical_CSS is `available` instead of `enabled` to avoid facing an error when loading the page with the module disabled and then enabling it to start generating C.CSS.

Reply:
Updated in fc609a3.